src/uct/ib/rc/accel/rc_mlx5_ep.c

```diff
@@ -131,6 +131,24 @@ uct_rc_mlx5_ep_am_short_inline(uct_ep_h tl_ep, uint8_t id, uint64_t hdr,
     return UCS_OK;
 }
 
+static ucs_status_t UCS_F_ALWAYS_INLINE uct_rc_mlx5_ep_am_short_iov_inline(
+        uct_ep_h tl_ep, uint8_t id, const uct_iov_t *iov, size_t iovcnt)
+{
+    UCT_RC_MLX5_EP_DECL(tl_ep, iface, ep);
+    UCT_RC_MLX5_CHECK_AM_SHORT(id, uct_rc_mlx5_hdr_t,
+                               uct_iov_total_length(iov, iovcnt), 0);
+    UCT_CHECK_IOV_SIZE(iovcnt, UCT_IB_MLX5_AM_ZCOPY_MAX_IOV,
```

Comment:
checking for `UCT_RC_MLX5_CHECK_AM_SHORT` should be enough, no need to rely on max IOV limit.